chromium/background.js

```diff
@@ -674,5 +682,56 @@ chrome.runtime.onMessage.addListener(function(message, sender, sendResponse){
     updateState();
   } else if (message.type == "remove_rule") {
     removeRule(message.object);
+  } else if (message.type == "import_settings") {
+    // This is used when importing settings from the options ui
+    import_settings(message.object).then(() => {
+      sendResponse(true);
+    });
   }
 });
+
+// Send a message to the embedded webextension bootstrap.js to get settings to import
+chrome.runtime.sendMessage("import-legacy-data", import_settings);
+
+/**
+ * Import extension settings (custom rulesets, ruleset toggles, globals) from an object
+ * @param settings the settings object
+ */
+async function import_settings(settings){
+  if(settings.changed){
+    // Load custom rulesets and add to storage
+    await new Promise(resolve => {
+      storage.set({"legacy_custom_rulesets": settings.custom_rulesets}, resolve);
+    });
+    load_legacy_custom_rulesets(settings.custom_rulesets);
+
+    // Load all the ruleset toggles into memory and store
+    let rule_toggle_promises = [];
+    for(let ruleset_name in settings.rule_toggle){
+      localStorage[ruleset_name] = settings.rule_toggle[ruleset_name];
+
+      (ruleset_name, ruleset_active => {
+        for(let host in all_rules.targets){
+          for(let ruleset of all_rules.targets[host]){
+            if(ruleset.name == ruleset_name){
+              ruleset.active = ruleset_active;
+              return;
+            }
+          }
+        }
+      })(ruleset_name, settings.rule_toggle[ruleset_name]);
+    }
+
+    // This is necessary in order to clear the cache for any previous ruleset lookups
+    all_rules.ruleCache = new Map();
+
+    // Set/store globals
+    await new Promise(resolve => {
+      storage.set({'httpNowhere': settings.prefs.http_nowhere_enabled}, resolve);
+    });
+    await new Promise(resolve => {
+      storage.set({'showCounter': settings.prefs.show_counter}, resolve);
+    });
+    isExtensionEnabled = settings.prefs.global_enabled;
```

Comment:
Did you see my comment about removing this since it will only work once, and is therefore inconsistent/confusing?